iree/compiler/Conversion/LinalgToSPIRV/VectorizeMemref.cpp

```diff
@@ -0,0 +1,357 @@
+//===- VectorizeMemref.cpp - Convert memref of scalar to vector------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Pass to convert memref into memref of vector.
+//
+//===----------------------------------------------------------------------===//
+#include "iree/compiler/Conversion/LinalgToSPIRV/Passes.h"
+#include "iree/compiler/Dialect/IREE/IR/IREEOps.h"
+#include "mlir/Dialect/StandardOps/IR/Ops.h"
+#include "mlir/Dialect/Vector/VectorOps.h"
+#include "mlir/IR/StandardTypes.h"
+
+constexpr int kVectorizationSizeInBits = 128;
+constexpr int kVecSize = kVectorizationSizeInBits / (sizeof(float) * 8);
+
+namespace mlir {
+namespace iree_compiler {
+/// Returns true of the type is a memref that can be vectorized to vector<4xi32>
+static bool isMemRefAndVectorizable(Type type) {
+  auto memrefType = type.dyn_cast<MemRefType>();
```

Comment:
Somewhere below the vec4 type used for replacement is f32. If this is integer kind you need to do a bit cast from int -> float. I am not sure that is handled in SPIR-V. Is this something you know works (i.e. the generated SPIR-V kernel runs correctly, or passes validation)

Reply:
Thanks!